python/tests/test_aac/context/test_language_context.py

```diff
@@ -46,7 +46,7 @@ def test_get_definitions_by_name(self):
 
         context.parse_and_load(VALID_AAC_YAML_CONTENT_SPACE_IN_NAME)
         definitions = context.get_definitions_by_name("Test Schema")
-        self.assertEqual(definitions[0].name, "Test Schema")
+        self.assertEqual(definitions[0].name.replace(" ", ""), "TestSchema")
```

Comment:
Interesting and again confused, you did a bug adjustment for allowing spaces in definition names, so not sure why this had to be altered?

Reply:
Because the names are retrieved in python form, which cannot have spaces.

Reply:
But then how is this test not breaking till now? The bug was addressing being able to have spaces and find them accordingly I had thought?

Reply:
I am not sure why its only now breaking, but I do remember having to do a similar fix for another test.  There is no way to have the method get_definitions_by_name return a name with a space in it.  You can give the method a space, but it will always return a dictionary where the name has no space, because it uses python names.

Reply:
Yeah, the bug you addressed about spaces in the names is what allowed for that line to be what is was previously. Not at all understanding why it would need to be changed now. 

https://github.com/DevOps-MBSE/AaC/pull/863/files

Reply:
> I am not sure why its only now breaking, but I do remember having to do a similar fix for another test. There is no way to have the method get_definitions_by_name return a name with a space in it. You can give the method a space, but it will always return a dictionary where the name has no space, because it uses python names.

So a similar check is made with the manual removal in the test for the definition parser, so was going to originally say that this shouldn't have to change then. But, that PR I linked introduced that check and the space handling in language context itself, and it passed then as a result. So, definitely not understanding what's happened...

Reply:
Gaaah, love when the editor decides to be funky about how it wants to categorize the line -.-

So, opposing the point of the test does not meet our needs. That test set was explicitly around evaluating can the name be found with  space in it. I think what it has now highlighted is that our way of addressing the spaces in a name may have been too simple/an incorrect approach. 

I was able to get this test to continue passing by tweaking the referenced definition name to be "Test Schema2" since "TestSchema" exists because of the other provided definition for testing. That let the individual test pass and the test suite of TestLanguageContext. I think that is the right way to tweak this test for right now to get it to pass within the pipeline and we need to flag a bug to take another look at handling names with spaces.

Reply:
Huh, I also tried that and now it is working in both places.  I am confused on what is happening here, I guess I missed something or misdiagnosed the issue.  It looks like it's fine now with the spaces included, I'm not sure why it seems so inconsistent.

Reply:
So we hit something similar to this in legacy to 0.4.0 about the context being odd to persist between tests sometimes. I think that was contributing somewhat, but I also think that since we added the handling to permitting spaces it was causing a name conflict with two `TestSchema` definitions. So, we need to file a bug or add a note on the associated test suite task to make suer to look into it more thoroughly. Once that is done, believe this PR is good to merge. Since you initially encountered the issue, would like you to file the bug or post a comment to the test suite issue, please.

Reply:
Bug is filed on Jira.